### ISSUE TITLE
Use consistent paths in code samples (accross all guides)

### ISSUE THREAD

**Issue by tweakch** [cardano-foundation/developer-portal#152]
It would make sense to agree on a consistent environment across guides

From what I've seen so far, it's *pretty consistent* but not everywhere... I find that I have to switch directory structure between different guides and my files started ending up everywhere... 

I would suggest the following setup

```text
/home/user/cardano
|_ /keys # from cardano-cli guides
|_ /wallets # from cardano-wallet guides
|_ /db
|_ /integrate-cardano-samples
  |_ /id-of-guide-1
    |_ draft.tx
    |_ draft.signed
  |_ /id-of-guide-2
  |_ ...
|_ /build-tx-metadata-samples
  |_ /id-of-guide-1
  |_ /id-of-guide-2
  |_ ...
``` 

This setup would allow all guides to first ask the user to check that he is in the correct directory (and let him `cd` into it)

```
$ pwd
/home/user/cardano/integrate-cardano-samples
```

and then for `cardano-cli` commands "reach up" for `.skeys` or `protocol.json`

```bash {3}
$ cardano-cli query utxo \
--testnet-magic 1097911063 \
--address $(cat ../keys/payment1.addr)
```

```bash
cardano-cli transaction sign \
--tx-body-file tx.draft \
--signing-key-file ../keys/payment1.skey \
--testnet-magic 1097911063 \
--out-file tx.signed
```

**Comment by tweakch** [comment]
I think there should be a *Setup environment* page that guides can reference and explains the rational behind the structure...

**Comment by katomm** [comment]
You are providing outstanding contributions here. I love the idea of doing a pwd before guides. 

Regarding directories, I am a little hesitant although I agree with you completely. For example, the "build with transaction metadata" was named "transaction metadata" in the beginning. Then later renamed for better user experience/SEO/marketing reasons. I'm afraid that might happen again and we'll always have to redirect the links.

**Comment by tweakch** [comment]
Thank you I believe this portal is one of the most important things for driving adoption and I'm happy to help.

I can totally see why this is difficult to maintain.

Maybe we could embed everything in some kind of a curriculum with a number system that can be extended and has a logic to it. 

Like courses in university.

Then users could `cd` into `<courseId>` and work from there.

I'm makong this up as I go... 😅

**Comment by tweakch** [comment]
As I re-read your comment I noticed a possible misunderstanding:

> I'm afraid that might happen again and we'll always have to ***redirect the links***.

Are you talking about the **file structure on the server**? I'm talking about the **local workspace**. and the **paths** used in the guides. I am also much more hesitant to change the structure on the server and re-link the entire guides all the time. Let me explain why I raised this issue: 

When people work through the guides, they are asked to

```bash
mkdir -p /home/user/cardano/keys
```

Some devs will already have a folder `~/cardano` with tons of stuff in there, but ok. They finish the guide and hopefully learned a lot. Then in the next guide we ask them to

```bash
mkdir -p /home/user/receive-ada-sample/keys
```

Can't they use `~/cardano/keys` from the last guide? Why not `mkdir -p ~/cardano/receive-ada-sample` and use `../keys/...` in the commands? Obviously they can and most of them will. I just think it's a little clumsy, thats all.

A lot of devs will rely on these guides while trying to wrap their heads around transactions and wallets and metadata and I think it's unfair to make it harder for them than it is. That said, people have brains and will figure out a system that works for them.

**Comment by rdlrt** [comment]
Just a point of reference, while this folder structure naming convention in this doco should possibly only be present for guide presented here, the proposed folder structure will already be different/unique than any other guides (for example, for [guild-ops](https://cardano-community.github.io/guild-operators/basics/#folder-structure), there is very different folder structure intentionally kept outside of home folder)

**Comment by katomm** [comment]
@tweakch you are right, at first I thought you were talking about the file structure of the docs as well.

**Comment by tweakch** [comment]
Very important point... the guides need to contain best practices from the community.

I must admit I'm out of my depth pretty quickly when it comes to those issues so I wouldn't want to design these environments. I'm a developer and follow what my sysadmin tells me to do.

That said, I'm very much willing to learn and understand the guild's setup - because I only write documentation for environments I understand.

**Comment by tweakch** [comment]
Just a thought... Maybe a docusaurus plugin could support us be consistent.

`docs` plugin already reports broken markdown links [here](https://github.com/facebook/docusaurus/blob/8bdb3da233caea7ed936bcd8013b1d50abc865f9/packages/docusaurus-plugin-content-docs/src/index.ts#L389)

We could write our own... quick and dirty one. Am I overthinking it? 🤣

**Comment by 0xBora** [comment]
The conversation in this issue has concluded, so it will now be closed.
